platform-sdk/swirlds-cli/src/main/java/com/swirlds/cli/logging/CssRuleSetFactory.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.cli.logging;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Utility class for turning CSS rules into a single CSS string, for inclusion in an HTML page.
+ */
+public class CssRuleSetFactory {
+    /**
+     * The rules that will be part of the output CSS string.
+     */
+    private final List<String> rules = new ArrayList<>();
+
+    /**
+     * Create a string representing a CSS rule
+     *
+     * @return the CSS rule string
+     */
+    @NonNull
+    private static String createRuleString(
+            @NonNull final String selector, @NonNull final List<CssDeclaration> declarations) {
+        return selector + " {\n"
+                + String.join(
+                        "\n",
+                        declarations.stream().map(CssDeclaration::toString).toList())
+                + "\n}";
+    }
+
+    /**
+     * Add a rule that will be part of the output CSS
+     *
+     * @param selector     the CSS selector
+     * @param declarations the CSS declarations
+     */
+    public void addRule(@NonNull final String selector, @NonNull final CssDeclaration... declarations) {
+        Objects.requireNonNull(selector);
+        Objects.requireNonNull(declarations);
+
+        rules.add(createRuleString(selector, Arrays.asList(declarations)));
+    }
+
+    /**
+     * Generate the CSS string representing all the rules that have been added
+     *
+     * @return the CSS string
+     */
+    @NonNull
+    public String generateCss() {
+        return String.join("\n", rules);
+    }
```

Comment:
A pattern I've used in a bunch of string generating utilities is to create methods with this form:

I wonder if it would be worth creating a simple interface called `Renderable` for these types of classes.

```
public interface Renderable {
    void render(@NonNull final StringBuilder sb);

    @NonNull
    default String render() {
            final StringBuilder sb = new StringBuilder();
            render(sb);
            return sb.toString();
    }
}
```

Lots of string concatenation for large strings can be inefficient, which is why I use string builders. When I am creating strings using multiple string utilites, they all render into the same string builder.

Reply:
Feel free to make this change a stand alone PR if you wish. But if it's a stand alone PR it should probably be worked soon-ish, since it will be a prerequisite for merging JTR.

Reply:
I made an issue to do this. https://github.com/hashgraph/hedera-services/issues/8334 

I'll prioritize once this PR merges